### PR TITLE
Remove acurl and use acurl_ng as the default

### PR DESCRIPTION
**Pull request by ryanlinnit-sky**
Remove old acurl implementation and rename acurl_ng -> acurl

### CODE REVIEW

File: acurl_ng/README.md [hunk 1]
```diff
@@ -5,20 +5,14 @@ It is an asynchronous wrapper around [libcurl](https://curl.se/libcurl/) which i
 
 ## Using Acurl_NG In Mite
 
-The current implementation of Acurl_NG is behind a feature toggle which defaults to using the old implementation of acurl. To switch over to using the new version of Acurl_NG within mite, the flag `MITE_CONF_enable_new_acurl_implementation="True"`
-
-The gateway into Acurl_NG is through the CurlWrapper (discussed in [Architectural Notes](#Architectural-Notes)) and requires an event loop being passed to it's constructor. Below is the mite implementation of acurl, using the aforementioned flag to switch between versions of acurl:
+The gateway into Acurl_NG is through the CurlWrapper (discussed in [Architectural Notes](#Architectural-Notes)) and requires an event loop being passed to it's constructor. Below is the mite implementation of acurl:
```
Comment (aecay) [inline acurl_ng/README.md:8]:
```suggestion
The gateway into Acurl_NG is through the CurlWrapper (discussed in [Architectural Notes](#Architectural-Notes)) and requires an event loop being passed to its constructor. Below is the mite implementation of acurl:
```

File: mite_http/__init__.py [hunk 2]
```diff
@@ -14,11 +16,8 @@ def __init__(self, session):
         self.additional_metrics = {}
 
     def __getattr__(self, attrname):
-        try:
-            r = object.__getattr__(self, attrname)
-            return r
-        except AttributeError:
-            pass
+        with suppress(AttributeError):
```
Comment (aecay) [inline mite_http/__init__.py:19]: nice!

**Review by aecay — APPROVED** [review]
two small comments.  also, the first sentence in the "architectural notes" section of the acurl_ng readme references the old implementation and can thus be removed.

overall the change lgtm :tada:

**Review by Matt2298 — APPROVED** [review]

**Comment by ryanlinnit-sky** [comment]
@aecay thanks :D Still a draft PR at the minute, I found some more references in `mite_http/__init__.py`, `setup.cfg`, and 
`tox.ini`. I'll get them sorted and hopefully merge this week.

File: Dockerfile [hunk 3]
```diff
@@ -2,32 +2,24 @@
...
 
-FROM python:3.8-alpine3.13
+FROM python:3.9-alpine3.13
 
 # py3-cryptography is added here as a means to get python cryptography onto the image (prebuilt) and without
 # need to install rust compiler
 RUN apk add --no-cache gnupg libressl tar ca-certificates gcc cmake make libc-dev coreutils g++ libzmq zeromq zeromq-dev git curl-dev libffi libffi-dev libbz2 bzip2-dev xz-dev libjpeg jpeg-dev py3-cryptography
 
-# This little bit of magic caches the dependencies in a docker layer, so that
-# rebuilds locally are not so expensive
```
[the remark this answers is not in the document]
Comment (ryanlinnit-sky) [reply Dockerfile:12]: We can do. I didn't know if that was just related to these lines:

```
COPY acurl/setup.cfg /acurl_ng-setup.cfg
COPY acurl_ng/setup.cfg /acurl-setup.cfg
```

Where it swapped the names of acurl_ng -> acurl and vice versa

[the remark this answers is not in the document]
Comment (ryanlinnit-sky) [reply Dockerfile]: Yeah that makes sense, added that comment back in

**Pull request merged**